modules/swagger-codegen/src/main/resources/Java/libraries/okhttp-gson/ApiClient.mustache

```diff
@@ -608,11 +608,11 @@ public class ApiClient {
      *
      * @param contentTypes The Content-Type array to select from
      * @return The Content-Type header to use. If the given array is empty,
-     *   JSON will be used.
+     *   or matches "any", JSON will be used.
      */
     public String selectHeaderContentType(String[] contentTypes) {
-        if (contentTypes.length == 0) {
-            return "application/json";
+        if (contentTypes.length == 0 || contentTypes[0].equals("*/*")) {
```

Comment:
@lwander we've actually implemented the check in `isJsonMime`: https://github.com/lwander/swagger-codegen/blob/a73ee037181933ee9040cc90383f2d4611a49c85/modules/swagger-codegen/src/main/resources/Java/libraries/okhttp-gson/ApiClient.mustache#L580